applications/StructuralMechanicsApplication/custom_utilities/multipoint_constraint_data.hpp

```diff
@@ -0,0 +1,246 @@
+TORT  OR OTHERWISE, ARISING  FROM, OUT  OF OR  IN CONNECTION  WITH THE
+SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+==============================================================================
+*/
+//
+//   Project Name:        Kratos
+//   Last Modified by:    $Author: Aditya Ghantasala $
+//   Date:                $Date: 14-03-2017 $
+//   Revision:            $Revision: 1.00 $
+//
+//
+
+#if !defined(CONSTRAINT_SLAVE_H_INCLUDED)
+#define CONSTRAINT_SLAVE_H_INCLUDED
+
+#include <vector>
+#include <map>
+#include <iostream>
+
+namespace Kratos
+{
+
+/** \brief Quaternion
+	* A simple class that implements the main features of quaternion algebra
+	*/
+class MpcData : public Flags
+{
+
+  public:
+    typedef Dof<double>* DofPointerType;
+    typedef Dof<double> DofType;
+
+    ///@name Life Cycle
+    ///@{
+
+    /**
+		Creates a Quaternion from its coefficients.
+		@param id : The ID of the DOF which is considered as a slave
+		*/
+    MpcData()
+    {
+        this->totalNumberOfMasterDOFs = 0;
+    }
+    /// Destructor.
+    virtual ~MpcData(){};
+
+    ///@}
+
+  public:
+    ///@name Operators
+    ///@{
+
+    ///@}
+
+  public:
+    ///@name Access
+    ///@{
+
+    /**
+		Adds a master and a corresponding weight to the current slave
+		@param slaveDOF the DOF number which is to be added
+		*/
+    void AddSlave(DofPointerType slaveDOF)
+    {
+
+        if (this->slaveDofMasterDataMap.count(slaveDOF) > 0)
+        {
+            std::cout << "AddSlave :: Slave DOF already in the map ... skipping." << std::endl;
+        }
+        else
+        {
+            slaveDofVector.push_back(slaveDOF);
+            masterData p;
+            this->slaveDofMasterDataMap[slaveDOF] = p;
+        }
+    }
+    /**
+		Adds a slave to the data structure
+		@param slaveDOF the slave DOF to which master DOF is to be added
+		@param masterDOF the DOF of the master DOF which is to be added
+		@param masterWeight the weight of the masterDOF
+		*/
+    void AddMaster(DofPointerType slaveDOF, DofPointerType masterDOF, double masterWeight)
+    {
+        this->slaveDofMasterDataMap[slaveDOF].masterDOFs.push_back(masterDOF);
+        this->slaveDofMasterDataMap[slaveDOF].masterWeights.push_back(masterWeight);
+    }
+
+    /**
+		Get the MasterDOFs vector for this slave
+		@return MasterDOFs vector for this slave
+		*/
+    const std::vector<DofPointerType> GetMasterDOFs(DofPointerType slaveDOF)
+    {
+        return this->slaveDofMasterDataMap[slaveDOF].masterDOFs;
+    }
+
+    /**
+		Get the weights for corresponding masters of a slave
+		@return weights for corresponding masters of a slave
+		*/
+    const std::vector<double> GetMasterWeightsForSlave(DofPointerType slaveDOF)
+    {
+        return this->slaveDofMasterDataMap[slaveDOF].masterWeights;
+    }
+
+    /**
+		Get the Total number of MasterDOFs in the current Node
+		@return Total number of MasterDOFs in the current Node
+		 */
+    const int GetTotalNumbeOfMasterDOFs()

```

Comment:
@RiccardoRossi Also in the example you gave, the slaves are different. so it will not be a problem. But in the case when the slaves are also the same, it will happen as I mentioned in reply to your above comment.